src/extract-functions.js

```diff
@@ -1,11 +1,17 @@
 import isFunction from './isFunction';
 
-export default function (...args) {
-  let result = [];
-  for (let i = 0; i < args.length; i += 1) {
-    const arg = args[i];
-    if (isFunction(arg)) result.push(arg);
-    else if (Array.isArray(arg)) result = result.concat(arg.filter(isFunction));
+export default function extractFunctions() {
+  const fns = [];
+
+  for (let i = 0; i < arguments.length; i += 1) {
+    const arg = arguments[i];
+
+    if (isFunction(arg)) {
+      fns.push(arg);
+    } else if (Array.isArray(arg)) {
+      fns.push(...(extractFunctions(...arg) || []));
+    }
   }
-  return result.length === 0 ? undefined : result;
+
+  return fns.length === 0 ? undefined : fns;
```

Comment:
I am aware that this was there before, but I am actually wondering why we are returning `undefined` here when no functions are found. It's not against spec and returning empty array will simply cause no iteration. 

It would also make the above recursive call better in the sense it wont need to create empty array knowing that this function always returns array.

@koresar ?